src/library_html5_webgpu.js

```diff
@@ -55,13 +55,14 @@ var LibraryHTML5WebGPU = {
     JsValStore.remove(id);
   },
 
-  // TODO(kainino0x): make it possible to actually create devices through webgpu.h
   emscripten_webgpu_get_device__deps: ['$WebGPU'],
   emscripten_webgpu_get_device: function() {
 #if ASSERTIONS
     assert(Module['preinitializedWebGPUDevice']);
 #endif
-    return WebGPU["mgrDevice"].create(Module['preinitializedWebGPUDevice']);
+    var device = Module['preinitializedWebGPUDevice'];
+    var deviceWrapper = { queueId: WebGPU.mgrQueue.create(device["queue"]) };
+    return WebGPU["mgrDevice"].create(device, deviceWrapper);
```

Comment:
Seem like we should be consistent about `WebGPU["mgrDevice"]` vs `WebGPU.mgrDevice`.

Probably worth keeping the closure extern list up-to-date so you can use the latter?

Reply:
I don't honestly remember why we do this in the first place, but whatever we intended I don't think it's working: the `mgrDevice` name is stable when written this way, `WebGPU` (as in `WebGPU.mgrDevice`) is not, so I don't think any external/EM_ASM code could be using this anyway. I'm going to go ahead and change them all to `.mgrDevice` with no extern.